midealocal/devices/e2/__init__.py

```diff
@@ -113,10 +113,12 @@ def set_attribute(self, attr: str, value: Any) -> None:
             DeviceAttributes.keep_warm,
             DeviceAttributes.current_temperature,
         ]:
+            old_protocol: str | bool | None = None
             if self._old_protocol is not None and self._old_protocol != "auto":
                 old_protocol = self._old_protocol
             else:
                 old_protocol = self.old_protocol()
+                raise TypeError("old_protocol is <bool> instead of <str>")

```

Comment:
Same question, why always raise `TypeError`?

Reply:
> Same question, why always raise `TypeError`?

As per my description: we don't know what is the right type, but is nearly always a `str` if not here.
I want to raise to see the error popping up (if ever it will) so we can act accordly.

Reply:
But this broke the code.

Reply:
No, is already broken.
A var that is expected to be a str in all comparisons and is instead a bool will anyway create a trace ;-)

Now we only surface the issue.
So as soon as we find a user with it, I can fix it properly.
But I'm 99% sure that one of the following the statement is true:

- this code block is never executed ( _old_protocol is always "auto" )
- no one use a E2 device

Reply:
See my comment below, I don't think current code is already broken.

Reply:
I don't agree, the reason is below. 

Note: why you stopped commenting here and putted a comment outside ? For others is more difficult to follow the discussion that way.

Reply:
Sorry, I was worried that the robot would always reply, let's continue outside.